File: 2024/day03_1.py
# pylint: disable=missing-module-docstring, missing-function-docstring


import re
from typing import Any


def solution(content: str) -> Any:
    return sum(int(a) * int(b) for a, b in re.findall(r"mul\((\d+),(\d+)\)", content))


def main() -> None:
    with open("day03.dat", "r", encoding="utf8") as file:
        print(solution(file.read().strip()))


if __name__ == "__main__":
    main()

File: 2024/day03-1.ts
function solution(content: string) {
  return content
    .matchAll(/mul\((\d+),(\d+)\)/g)
    .map(([_, a, b]) => Number(a) * Number(b))
    .reduce((acc, cur) => acc + cur, 0);
}

Deno.readTextFile("day03.dat")
  .then((content) => content.trim())
  .then(solution)
  .then(console.log);
